xds/src/test/java/io/grpc/xds/LookasideLbTest.java

```diff
@@ -243,80 +185,183 @@ public int getPriority() {
 
       @Override
       public String getPolicyName() {
-        return "supported1";
+        return "round_robin";
       }
 
       @Override
       public LoadBalancer newLoadBalancer(Helper helper) {
-        return mock(LoadBalancer.class);
+        String localityName = helper.getAuthority();
+        childHelpers.put(localityName, helper);
+        LoadBalancer balancer = mock(LoadBalancer.class);
+        childBalancers.put(localityName, balancer);
+        return balancer;
       }
     });
+  }
+
+  @After
+  public void tearDown() {
+    lookasideLb.shutdown();
 
+    for (LoadBalancer childBalancer : childBalancers.values()) {
+      verify(childBalancer).shutdown();
+    }
+
+    if (withBootstrap) {
+      assert !withXdsClientPoolAttributes;
```

Comment:
`assert`? What is this doing?

Reply:
This asserts that `withBootstrap` and `withXdsClientPoolAttributes` should not both be true, in order to prevent human error that sets up bootstrap and uses XdsClientPoolAttributes in the same test.

Reply:
Using flags are your test implementation. You should not "assert" your test implementation. Otherwise, test failures are likely to be mixed for code under test and test code. Ideally, test implementation should be simple and straightforward (e.g., call some code under test and verify the outcome) such that itself is unlikely to have bugs. Tests are based on the premise that test code contains no bug. Thus, you should ensure the test implementation does not contain bugs, not test should not assert itself. A better approach may be using [`Assume`](https://junit.org/junit4/javadoc/4.12/org/junit/Assume.html), such as the example used in [`AbstractInteropTest`](https://github.com/grpc/grpc-java/blob/04924d07b332e80b2253ab4c182380c2a7103e6f/interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java#L1069). Here, just a flag for different subtests, not bother to be overkilled. But anyway, I don't think you should have `assert` here.